microclient.go

```diff
@@ -0,0 +1,15 @@
+// microclient provides opinionated helpers for REST clients.
+package microclient
+
+import (
+	"context"
+
+	"github.com/go-resty/resty"
+)
+
+// Do takes a resty request function and executes it, adding metrics.
+func Do(ctx context.Context, requestFunc func(string) (resty.Response, error)) (resty.Response, error) {
+	// TODO: add metrics.
+	// TODO: actually perform the request.
+	return resty.Response{}, nil
+}
```

Comment:
I'm not a fan of writing wrappers over wrappers. But when it's helpful why not preserve the APIs to make life easier. This repo could be named `resty` and add define `R` func https://godoc.org/github.com/go-resty/resty#R as in the resty API. Then using the package would be just a matter of switching import and the API stays familiar. AFAIK this is a common practice.

Reply:
We would basically need to rewrite the whole resty package. There are no interfaces. We would need to wrap everything on our own which is what you do not like. Joe and I talked about this and couldn't come up with a better idea.

Reply:
> We would need to wrap everything on our own which is what you do not like

I would like that actually. We don't have to wrap everything. Just the parts we need.